pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -202,20 +202,36 @@ def targets(self) -> tuple[KCFG.Node, ...]:
     class Edge(EdgeLike):
         source: KCFG.Node
         target: KCFG.Node
-        depth: int
-        rules: tuple[str, ...]
+        depths: tuple[int, ...]
+        rules_list: tuple[tuple[str, ...], ...]
+        csubsts: tuple[CSubst, ...]
```

Comment:
I'm not sure we can manage a breaking change of this scale across our stack at this point.

Is there a reasonable way to merge edges _with_ loss of information? (E.g. take the the maximum `depth`, etc.)

Reply:
I’ve added two properties to mitigate this issue, ensuring that previous access to `depth` and `rule` continues to work as expected.

https://github.com/runtimeverification/k/blob/0b3669fcf65a69b19d1195003cf9053519e0f217/pyk/src/pyk/kcfg/kcfg.py#L209-L215

Reply:
`to_dict`  and `from_dict` still changes though, and as implemented, proofs written before the change will no longer be usable with this version.

We could still
1. Update the old format on the first read, or
2. Support both formats

but we should really think about if adding this complexity is worth it right now.

Reply:
If losing information is not an option for your use case, you could still add a new class `MergedEdge(EdgeLike)` that contains a `tuple[Edge, ...]`.

Reply:
Thank you for your suggestion! I think introducing a new class is a good idea. It won’t disrupt the existing proofs and will preserve the information that would otherwise be lost. However, we’ll still need to modify KCFGShow and KCFGViewer to adapt to this change, but the adjustments required will be much smaller.

Reply:
Moreover, we need to change the `as_rule` and `as_rules` to make this new Edge class work correctly for the CSE.

Reply:
If that implementation is better, I will close this PR and work that then. What do you think?

Reply:
I like the `MergeEdge` idea!